apps/web/app/features/stacking/start-pooled-stacking.tsx

```diff
@@ -0,0 +1,149 @@
+
+interface StartPooledStackingProps {
+  poolName: PoolName;
+}
+
+export function StartPooledStacking({ poolName }: StartPooledStackingProps) {
+  const { client } = useStackingClient();
+  const { stxAddress } = useLeatherConnect();
+
+  if (!stxAddress || !client) {
+    return 'You should connect STX wallet';
+  }
+  if (!client) {
+    return 'Expected client to be defined';
+  }
+
+  return <StartPooledStackingLayout poolName={poolName} />;
+}
+
+const initialStackingFormValues: Partial<StackingFormValues> = {
+  amount: '',
+  numberOfCycles: 1,
+  poolAddress: '',
+};
+
+interface StartPooledStackingLayoutProps {
+  poolName: PoolName;
+}
+
+// eslint-disable-next-line no-empty-pattern
+function StartPooledStackingLayout({}: StartPooledStackingLayoutProps) {
+  const { stxAddress, btcAddressP2wpkh } = useLeatherConnect();
+  const { network, networkInstance } = useStacksNetwork();
+  const poxContracts = getPoxContracts(network);
+
+  const getSecondsUntilNextCycleQuery = useGetSecondsUntilNextCycleQuery();
+
+  const [contractAddress, contractName] = getPoxContracts(network)['Pox4'].split('.');
+
+  const getAllowanceContractCallersFastPoolQuery = useGetAllowanceContractCallersQuery({
+    contractAddress,
+    contractName,
+    callingContract: poxContracts['WrapperFastPool'],
+    senderAddress: stxAddress ? stxAddress.address : null,
+    network,
+  });
+
+  const getAllowanceContractCallersRestakeQuery = useGetAllowanceContractCallersQuery({
+    contractAddress,
+    contractName,
+    callingContract: poxContracts['WrapperRestake'],
+    senderAddress: stxAddress ? stxAddress.address : null,
+    network,
+  });
+
+  const getAllowanceContractCallersOneCycleQuery = useGetAllowanceContractCallersQuery({
+    contractAddress,
+    contractName,
+    callingContract: poxContracts['WrapperOneCycle'],
+    senderAddress: stxAddress ? stxAddress.address : null,
+    network,
+  });
+
+  const [hasUserConfirmedPoolWrapperContract, setHasUserConfirmedPoolWrapperContract] =
+    useState<PoolWrapperAllowanceState>({});
+
+  const derivedAllowanceState = useMemo(
+    () => ({
+      [networkInstance]: {
+        [poxContracts['Pox4']]: true,
+        [poxContracts['WrapperFastPool']]:
+          getAllowanceContractCallersFastPoolQuery?.data?.type === ClarityType.OptionalSome,
+        [poxContracts['WrapperRestake']]:
+          getAllowanceContractCallersRestakeQuery?.data?.type === ClarityType.OptionalSome,
+        [poxContracts['WrapperOneCycle']]:
+          getAllowanceContractCallersOneCycleQuery?.data?.type === ClarityType.OptionalSome,
+      },
+    }),
+    [
+      networkInstance,
+      poxContracts,
+      getAllowanceContractCallersFastPoolQuery?.data?.type,
+      getAllowanceContractCallersRestakeQuery?.data?.type,
+      getAllowanceContractCallersOneCycleQuery?.data?.type,
+    ]
+  );
+
+  useEffect(() => {
+    setHasUserConfirmedPoolWrapperContract(derivedAllowanceState);
+  }, [derivedAllowanceState]);
```

Comment:
I don't understand why we need this effect? `useMemo` updates with its dependencies if it changes?

Reply:
yeah... no need in it